packages/components/src/internal/components/picklist/ChoosePicklistModal.tsx

```diff
@@ -475,56 +503,75 @@ export const ChoosePicklistModal: FC<ChoosePicklistModalProps> = memo(props => {
                 setError(resolveErrorMessage(e) ?? 'Failed to retrieve picklists.');
             }
             setItemsLoading(LoadingState.LOADED);
+            if (useSnapshotSelection) {
+                if (!queryModel?.isLoadingSelections) {
+                    try {
+                        await setSnapshotSelections(queryModel.selectionKey, [...queryModel.selections]);
+                    }
+                    catch (reason) {
+                        console.error("There was a problem loading the filtered selection data. Your actions will not obey these filters.", reason);
+                    }
+                    setSelectionsLoading(LoadingState.LOADED);
+                }
+            }
+            else {
+                setSelectionsLoading(LoadingState.LOADED);
+            }
         })();
-    }, []);
+    }, [queryModel?.selectionKey, queryModel?.selections, queryModel?.isLoadingSelections, useSnapshotSelection]);
```

Comment:
does adding in these dependencies to the useEffect mean that the `await getPicklistsForInsert()` will get called multiple times? Do we need/want to add an if statement around that so that it only loads and calls `setItems()` once?

Reply:
Done